mutation_compactor.hh

```diff
@@ -292,11 +297,15 @@ public:
         , _last_dk({dht::token(), partition_key::make_empty()})
         , _last_pos(position_in_partition::for_partition_end())
         , _collector(std::make_unique<mutation_compactor_garbage_collector>(_schema))
+        // We already have a validator for compaction in the sstable writer, no need to validate twice
+        , _validator("mutation_compactor for compaction", _schema, mutation_fragment_stream_validation_level::none)
```

Comment:
This constructs a string in mutation_fragment_stream_validating_filter, adding an allocation to all reads. Please follow up with changing that to a string_view (and verifying all callers indeed pass a constant; you can append the "sv" literal operator as a marker).

Reply:
Some callers pass in string created with `format()`. We can do the usual trick of having a string and string_view in the class to avoid allocating when possible.

Reply:
Followup PR is here: https://github.com/scylladb/scylladb/pull/12042